src/index.ts

```diff
@@ -613,7 +613,16 @@ class MapVisualizer {
         // We need to create pseudo structures with 0 atoms so that the indexer
         // knows how to associate structure id & point id on the map. The
         // structure id is used by the info panel to create the slider
-        const n_structure = dataset.properties[Object.keys(dataset.properties)[0]].values.length;
+
+        // unless all properties have atom targets, we assume the map is meant to represent structures
+        let n_structure = 0;
+        for (const key of Object.keys(dataset.properties)) {
+            n_structure = dataset.properties[key].values.length;
+            if (dataset.properties[key].target === 'structure') {
+                break;
+            }
+        }
```

Comment:
Why is this? The default for the other viewer is to use per-atom properties if environments are present, otherwise use per-structure properties.

Reply:
I had this problem where a dataset of structures that had stray environment properties in it was completely broken when showing a map-only viewer. When we do a combined view it can use info on whether environments are in, here it relies only on counting the number of entries in the first property. For sure it was a bug, whether this is the right way to fix it, I'm open to alternative strategies.

Reply:
Do you have the notebook around? It might be possible to fix this by detecting the presence/absence of the `environments` parameter instead